index.bs

```diff
@@ -2430,45 +2466,90 @@ To <dfn>check if an [=attribution source=] exceeds the time-based destination li
+        <dl class="switch">
+        : "<code>[=rate-limit scope/source=]</code>"
+        :: Set |record|'s [=attribution rate-limit record/deactivated for unexpired destination limit=] to
+            true if |sourcesToDelete| [=set/contains=] |record|'s [=attribution rate-limit record/entity ID=].
+        : "<code>[=rate-limit scope/aggregatable-attribution=]</code>"
+        :: [=set/Remove=] |record| from the [=attribution rate-limit cache=] if
+            |deletedAggregatableReports| [=set/contains=] |record|'s [=attribution rate-limit record/entity ID=].
+
+        </dl>
+
+To <dfn>get sources to delete for the unexpired destination limit</dfn> given an
+[=attribution source=] |source|, run the following steps:
+1. Let |records| be a new [=list=].
+1. [=set/iterate|For each=] [=attribution rate-limit record=] |record| of the [=attribution rate-limit cache=]:
+    1. If |record|'s [=attribution rate-limit record/scope=] is not "<code>[=rate-limit scope/source=]</code>", [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/deactivated for unexpired destination limit=] is true, [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/source site=] and |source|'s [=attribution source/source site=] are not equal, [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/reporting origin=] and |source|'s [=attribution source/reporting origin=] are not [=same site=], [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/expiry time=] is less than or equal to |source|'s [=attribution source/source time=], [=iteration/continue=].
+    1. [=Assert=]: |record|'s [=attribution rate-limit record/destination limit priority=] is not null.
+    1. [=Assert=]: |record|'s [=attribution rate-limit record/entity ID=] is not null.
+    1. [=list/Append=] the [=tuple=] (|record|'s [=attribution rate-limit record/destination limit priority=],
+        |record|'s [=attribution rate-limit record/time=], |record|'s [=attribution rate-limit record/attribution destination=],
+        |record|'s [=attribution rate-limit record/entity ID=]) to |records|.
+1. [=set/iterate|For each=] [=site=] |destination| of |source|'s [=attribution source/attribution destinations=]:
+    1. [=list/Append=] the [=tuple=] (|source|'s [=attribution source/destination limit priority=],
+        |source|'s [=attribution source/source time=], |destination|, |source|'s [=attribution source/source identifier=])
+        to |records|.
+1. [=list/sort in descending order|Sort=] |records| in descending order, with |a| less than |b| if the following steps return true:
+    1. If |a|[0] is less than |b|[0], return true.
+    1. If |a|[0] is greater than |b|[0], return false.
+    1. If |a|[1] is less than |b|[1], return true.
+    1. If |a|[1] is greater than |b|[1], return false.
+    1. If <a href="https://html.spec.whatwg.org/multipage/origin.html#serialization-of-a-site">serialized</a> |a|[2]
```

Comment:
I was surprised to see that we need to explicitly serialize the site here in order to be able to compare it with another site, because I thought that our use of sets of sites elsewhere in the spec implied that sites already had an ordering operation.

But then I noticed that Infra sets (which are always "ordered sets") don't actually enforce any ordering beyond the ordering of their underlying list, based on [this example](https://infra.spec.whatwg.org/#example-set-replace). In other words, the Infra set and map types aren't, in a Java analogy, `TreeSet` or `TreeMap`, but rather `LinkedHashSet` or `LinkedHashMap`.

So my comment here isn't an issue with this PR, but I think we need to audit our use of sets and maps again. Related to #1193.